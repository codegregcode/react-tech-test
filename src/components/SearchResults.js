import React from "react";
import "../styles/search-results.css";

const SearchResults = ({ results }) => {
  if (!results.length) {
    return <p>Search the universe using the search bar above!</p>;
  } else {
    return (
      <div className="search-results-container">
        {results.map((image, index) => (
          <img
            className="card-image"
            key={index}
            src={image.src}
            alt={image.alt}
          />
        ))}
      </div>
    );
  }
};

export default SearchResults;
