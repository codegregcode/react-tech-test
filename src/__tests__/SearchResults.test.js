import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const validProps = ["image1", "image2", "image3", "image4"];
  const { asFragment } = render(<SearchResults results={validProps} />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("returns images", () => {
    render(<SearchResults results={validProps} />);

    expect(screen.getAllByRole("img")).toHaveLength(4);
  });
});
