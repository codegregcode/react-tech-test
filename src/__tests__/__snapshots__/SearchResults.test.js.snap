// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results-container"
  >
    <img
      class="card-image"
    />
    <img
      class="card-image"
    />
    <img
      class="card-image"
    />
    <img
      class="card-image"
    />
  </div>
</DocumentFragment>
`;
